### ISSUE TITLE
Opt-in `Share Usage Data` option

### ISSUE THREAD

**Issue by c2xusnpq6** [AdguardTeam/AdGuardHome#3766]
I don’t see `Willing to share usage data` on the management page. Is this because of default sharing? Or? 

Because I want to contribute data to improve the overall user experience, thank you!

**Comment by rodalpho** [comment]
My feeling is the type of person who goes to the trouble to host his own adblocking DNS server on their LAN is the same type of dude who wouldn't turn on telemetry. I know I certainly wouldn't! But since you're on the side of the angels with it being opt-in, I'm all for it.

**Comment by c2xusnpq6** [comment]
> My feeling is the type of person who goes to the trouble to host his own adblocking DNS server on their LAN is the same type of dude who wouldn't turn on telemetry. I know I certainly wouldn't! But since you're on the side of the angels with it being opt-in, I'm all for it.

You can see the "Optimized Version" button on AdGuard. It may contain 80% or 90% of the rules from the original version.

But the question arises, how do you know which rules are commonly used? Which ones are not?

**Comment by c2xusnpq6** [comment]
@rodalpho > I hope Adguard Home can display multiple related buttons, such as whether to send an anonymized IP message?
>
> Is there any other data from the current server to be sent?
>
> These are hoped that Adguard can be listed in detail and available for users to choose.
>
> I also hope that this feature can support mobile and PC version, thank you!
>
>Both PC and mobile version now only have "Agree to Share Usage Data" and "Disagree to Send Usage Data", but there is no detailed description of which data will be sent...

For software that claims to pay special attention to user privacy, this will not look good...

**Comment by Disqu3-mirOir-qt** [comment]
> My feeling is the type of person who goes to the trouble to host his own adblocking DNS server on their LAN is the same type of dude who wouldn't turn on telemetry. I know I certainly wouldn't! But since you're on the side of the angels with it being opt-in, I'm all for it. 

i agree i use adguard home on my owns server i don,t wan,t telemetry seem off , in this case or telemetry added i do it owns build with telemetry code part removed

**Comment by c2xusnpq6** [comment]
> > My feeling is the type of person who goes to the trouble to host his own adblocking DNS server on their LAN is the same type of dude who wouldn't turn on telemetry. I know I certainly wouldn't! But since you're on the side of the angels with it being opt-in, I'm all for it. 
> 
> i agree i use adguard home on my owns server i don,t wan,t telemetry seem off , in this case or telemetry added i do it owns build with telemetry code part removed 

If this is the case, can you ask them to release a version without the associated tracking code for each new version?

**Comment by ameshkov** [comment]
> You may be concerned that those functions can be turned on remotely?

I don't think this is a valid concern. One can always validate that it works as it's said and cannot be enabled remotely since the code is open.

**Comment by Disqu3-mirOir-qt** [comment]
> > > My feeling is the type of person who goes to the trouble to host his own adblocking DNS server on their LAN is the same type of dude who wouldn't turn on telemetry. I know I certainly wouldn't! But since you're on the side of the angels with it being opt-in, I'm all for it. 
> > 
> > i agree i use adguard home on my owns server i don,t wan,t telemetry seem off , in this case or telemetry added i do it owns build with telemetry code part removed 
> 
> You may be concerned that those functions can be turned on remotely?
> 
> If this is the case, can you ask them to release a version without the associated tracking code for each new version?
> The problem is solved~

not , i said if telemetey added i do my owns build adguardHome(go) with telemetry part removed

**Comment by rodalpho** [comment]
Well, that's your prerogative. I have no problems with an opt-in telemetry switch. These metrics can be genuinely useful to improve software, they have a legitimate reason to exist. 

Thing is, I personally wouldn't opt-in and I doubt many people who run this type of software would either. So it may not be worth their effort to implement.

**Comment by Disqu3-mirOir-qt** [comment]
> Well, that's your prerogative. I have no problems with an opt-in telemetry switch. These metrics can be genuinely useful to improve software, they have a legitimate reason to exist. 
> 
> Thing is, I personally wouldn't opt-in and I doubt many people who run this type of software would either. So it may not be worth their effort to implement.

I agree with the telemetry in the friendly user software for example in the new server dns of adguard I have no problem, but on self-hosted software it loses all the interest to do self-hosting if I hosted adguardHome on my vps is not to have telemetry, adguard already has a github to improve their software, 

I still prefer to pay a license / monthly subscription instead of telemetry

**Comment by c2xusnpq6** [comment]
@ainar-g @ameshkov or can you give users who are willing to share anonymous data the benefit of the doubt?

Anti-virus software is also this way to collect more virus data, there is a limited use of the free version?

**Comment by ameshkov** [comment]
Hey all, as I've said, we'll only consider implementing this if there are enough people voting for this feature request.

**Comment by Disqu3-mirOir-qt** [comment]
> @ainar-g @ameshkov or can you give users who are willing to share anonymous data the benefit of the doubt?
> 
> Anti-virus software is also this way to collect more virus data, there is a limited use of the free version?
> 
> The beta and nightly version will more obviously prompt users to share anonymous data or not?

anti-virus software is easy to install, no anti-virus software (e.g. maybe clamAV) have a self-hosting system

anonymous data is a marketing term to give an impression of trust , reverse data when data big easy

https://gizmodo.com/researchers-reveal-that-anonymized-data-is-easy-to-reve-1836629166

**Comment by c2xusnpq6** [comment]
> > @ainar-g @ameshkov or can you give users who are willing to share anonymous data the benefit of the doubt?
> > 
> > Anti-virus software is also this way to collect more virus data, there is a limited use of the free version?
> > 
> > The beta and nightly version will more obviously prompt users to share anonymous data or not?
> 
> anti-virus software is easy to install, no anti-virus software (e.g. maybe clamAV) have a self-hosting system
> 
> anonymous data is a marketing term to give an impression of trust , reverse data when data big easy
> 
> https://gizmodo.com/researchers-reveal-that-anonymized-data-is-easy-to-reve-1836629166

So what you're saying is that they need to ensure complete security to do so?

You need them to move the head office to Switzerland? Like Kaspersky did?🤔😃👍

**Comment by rodalpho** [comment]
I agree the concept of an anonymous user is flat-out incorrect, as even when tagged to a generic UUID that data can be correlated with other activity to track down your true identity. However, it's possible to collect data truly anonymously via differential privacy, where data is aggregated such that you can't point to any individual user's activity, even an "anonymous" user. These algorithms can be mathematically proven and are an interesting rabbit-hole to explore on a hobbyist level, but I can't imagine Adguard would go to the trouble to implement them.

**Comment by IRainman** [comment]
I hopefully share my data from this setup #5819 , because it's a beautiful test case for memory corruption or race conditions.
